### PR TITLE
torch.hub: add get_dir/set_dir type hints

### PR DESCRIPTION
**Pull request by adamjstewart**
cc @ezyang @malfet @xuzhao9 @gramster

### CODE REVIEW

File: torch/hub.py [hunk 1]
```diff
@@ -408,7 +408,7 @@ def get_dir():
     return os.path.join(_get_torch_home(), "hub")
 
 
-def set_dir(d):
+def set_dir(d: Union[str, os.PathLike]) -> None:
```
Comment (Skylion007) [inline torch/hub.py:411]: I think this can technically be a bytes string too, (although I presume that would be really rare).

Comment (adamjstewart) [reply torch/hub.py:411]: I just checked the rest of the repo and `Union[str, os.PathLike]` is the most commonly used way to represent paths. Not opposed to adding `bytes`, just wanted to match the rest of the repo.

**Comment by adamjstewart** [comment]
Does this PR need release notes, or is it not user facing?

**Review by Skylion007 — APPROVED** [review]

**Comment by adamjstewart** [comment]
Needs https://github.com/pytorch/pytorch/pull/134875, let me rebase

**Pull request closed without merging**